parser_library/src/context/hlasm_context.h

```diff
@@ -263,6 +263,12 @@ class hlasm_context
         if (auto glob = globals_.find(id); glob != globals_.end())
         {
             set_sym_ptr var = std::dynamic_pointer_cast<set_symbol<T>>(glob->second);
+            if (var == nullptr)
```

Comment:
can basically the same issue happen in the previous if statement?

As far as I can tell, the only way this function can fail is due to type mismatch. So, I'd suggest returning nullptr in that case and let the caller handle the situation.

Reply:
is create_local_variable affected as well?

Reply:
- Technically it could, however, `prepare_GBL_LCL` takes care of these cases and issues `E051` error. But maybe these checks can be moved here
- That's true, it can only happen due to type mismatch. However, if `nullptr` is returned here it would result in a small inconsistency with `LCL` symbols as these don't throw away the variable from scope and error `E051` is issued in a similar case. Additionally, the PR behavior simulates the original one more closely. I'm not saying I'm against changing this in a proposed way I just feel there is a need for further discussion
- `create_local_variable`: When dealing with local variables there are checks in place that take place before `create_local_variable` is reached and they result either in `E013` or `E051` errors